src/position.cpp

```diff
@@ -276,6 +317,9 @@ Position& Position::set(const string& fenStr, bool isChess960, StateInfo* si, Th
   // 5-6. Halfmove clock and fullmove number
   ss >> std::skipws >> st->rule50 >> gamePly;
 
+  if (st->rule50 > 100)
```

Comment:
I think this is hardly an issue since a user can work around this and submit a fen of counter 99, 2 moves earlier and then get his way through to the same fen we rejected.
On a different note If this is really relevant I think what should reject is negative values for integers instead.